src/astro/files/base.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Optional, Union
+
+import pandas as pd
+import smart_open
+
+from astro.constants import FileType
+from astro.files.locations import create_file_location
+from astro.files.type import create_file_type
+
+
+class File:
+    """Handle all file operations, and abstract away the details related to location and file types.
+    Intended to be used within library.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        conn_id: Optional[str] = None,
+        filetype: Union[FileType, None] = None,
+    ):
+        """Init file object which represent a single file in local/object stores
+
+        :param path: Path to a file in the filesystem/Object stores
+        :param conn_id: Airflow connection ID
+        """
+        self.location = create_file_location(path, conn_id)
+        self.type = create_file_type(path=path, filetype=filetype)
+
+    @property
+    def path(self):
+        return self.location.path
+
+    @property
+    def size(self) -> int:
+        """
+        Return the size (bytes) of the given file.
+
+        :return: File size in bytes
+        :rtype: int
+        """
+        size: int = self.location.size
+        return size
+
+    def is_binary(self) -> bool:
+        """
+        Return a FileType given the filepath. Uses a naive strategy, using the file extension.
+
+        :return: True or False
+        :rtype: bool
+        """
+        result: bool = self.type.name == FileType.PARQUET
+        return result
+
+    def write_from_dataframe(self, df: pd.DataFrame) -> None:
```

Comment:
I was initially confused with the name. What do you think of `create_from_dataframe`?

I think the confusion is that usually when we have `write,` we'd expect a `to`. In this case, since the class is a file, it would be redundant to say `write_file_from_dataframe`, so I can fully understand why you dropped the "file".

Reply:
Done.